portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java

```diff
@@ -1,18 +1,50 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.*;
 
 /** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
 @WebServlet("/hello")
 public class HelloWorldServlet extends HttpServlet {
-
+    
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Here's a fetch response!");
+  public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+    ArrayList<String> randFacts = new ArrayList<String>();
+    randFacts.add("I was born somewhere between Laos and Thailand while my parents were en route to a refugee camp.");
+    randFacts.add("I recently moved to Fresno, CA from Visalia, CA and I'm living with 2 dogs, Max and Mister.");
+    randFacts.add("I love dogs (and i'm also allergic to them).");
+    randFacts.add("My favorite food is shrimp. I will eat any dish if it's included.");
+    String json = convertToJson(randFacts);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+    private String convertToJson(ArrayList<String> randFacts) {
+    String json = "{";
+    json += "\"fact1\": ";
+    json += "\"" + randFacts.get(0) + "\"";
+    json += ",  ";
+    json += "\"fact2\": ";
+    json += "\"" + randFacts.get(1) + "\"";
+    json += ",  ";
+    json += "\"fact3\": ";
+    json += "\"" + randFacts.get(2) + "\"";
+     json += ",  ";
+    json += "\"fact4\": ";
+    json += "\"" + randFacts.get(3) + "\"";
+    json += "}";
+    return json;
+  }
+
+    private String convertToJsonUsingGson(ArrayList<String> randFacts) {
+        final Gson gson = new Gson();
+        final String json = gson.toJson(randFacts);
```

Comment:
optional 🤷 : no need here to capture the response in a variable if you immediately return it.

Exs:

`return gson.toJson(data);` or even keep a `Gson` object as a static field on the Servlet like 

```
public class FactServlet extends HttpServlet {
  public static final Gson gson = new Gson();

@Override  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
  //...
  String json = gson.toJson(data);
```
which allows you to get rid of the function altogether.


Side note: Don't let me discourage you from using variables like this. Local variables, especially non-necessary ones, are great for making your code more readable. A toy example:

```
// before 
if (clock.getHours() > 12 && clock.isPm() && clock.getHours() < 5) {
  return true;
}
...
```

```
// after
boolean isAfterNoon = clock.getHours() > 12 && clock.isPm() && clock.getHours() < 5;

if (isAfternoon) {
  return true;
}
```

Reply:
done. changes will be on the next commit